primitives/trie/src/node_codec.rs

```diff
@@ -54,9 +54,7 @@ impl<'a> ByteSliceInput<'a> {
 
 impl<'a> Input for ByteSliceInput<'a> {
 	fn remaining_len(&mut self) -> Result<Option<usize>, codec::Error> {
-		let remaining =
-			if self.offset <= self.data.len() { Some(self.data.len() - self.offset) } else { None };
-		Ok(remaining)
+		Ok(Some(self.data.len().saturating_sub(self.offset)))
```

Comment:
fix code, does not look to me like switching could be an issue :+1:

Reply:
The behavior before was wrong. `remaining_len` should only return `None` if it doesn't know what the remaining length is.